### PR TITLE
#26rng - As a MGD I want to see a list of Ms

### PR DESCRIPTION
**Pull request by ThijsvanDam**
You can see a list of M's.
Didn't do the implementation of a scrollable list on more than 3 M's yet. Didn't feel really necessary.

### CODE REVIEW

File: core/src/com/marama/view/entities/EntityManagerSingleton.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+public class EntityManagerSingleton extends AssetManager {
+    private static EntityManagerSingleton instance = new EntityManagerSingleton();
+
+    // Since in AssetManager there is no good way to get all assets of a certain type, we have to remember our maramafications here.
+    final ObjectMap<String, Maramafication> maramafications = new ObjectMap();
+
+    public ObjLoader objLoader = new ObjLoader();
+
+    /**
+     * An instance of {@link AssetManager}, but it's also able to load maramafications.
```
Comment (LuneCoding) [inline core/src/com/marama/view/entities/EntityManagerSingleton.java]: An instance of Assetmanager that is also able to load maramafications.

Might be better?

Comment (wilbo) [reply core/src/com/marama/view/entities/EntityManagerSingleton.java]: Sounds better, on top of this, I don't think you should put the name of the design pattern in the class name. I would just call it EntityManager. The description/javaDoc says enough about what this class is and does.

**Review by wilbo — CHANGES_REQUESTED** [review]
Rename the EntityManagerSingleton class, other than that it is good.

File: core/src/com/marama/view/util/MaramaficationLoader.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.maramafication = null;
+        return maramafication;
+    }
+
+    @Override
+    public Array<AssetDescriptor> getDependencies(String fileName, FileHandle file, MaramaficationLoader.MaramaficationLoaderParameter parameter) {
+        return null;
+    }
+
+    static public class MaramaficationLoaderParameter extends AssetLoaderParameters<Maramafication> {
```
Comment (LuneCoding) [inline core/src/com/marama/view/util/MaramaficationLoader.java:66]: This is a todo, right? Please make a backlog item for it!

File: core/src/com/marama/view/util/MaramaficationLoader.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    public Maramafication loadSync(AssetManager manager, String fileName, FileHandle file, MaramaficationLoader.MaramaficationLoaderParameter parameter) {
+        // Get the maramafication that is currently loaded, and clear it
+        Maramafication maramafication = this.maramafication;
+        this.maramafication = null;
+        return maramafication;
+    }
+
+    @Override
+    public Array<AssetDescriptor> getDependencies(String fileName, FileHandle file, MaramaficationLoader.MaramaficationLoaderParameter parameter) {
+        return null;
```
Comment (LuneCoding) [inline core/src/com/marama/view/util/MaramaficationLoader.java:63]: This is a todo, right? Please make a backlog item for it!

File: core/src/com/marama/view/util/MaramaficationLoader.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        jsonReader = new JsonReader();
+    }
+
+    @Override
+    public void loadAsync(AssetManager manager, String fileName, FileHandle file, MaramaficationLoader.MaramaficationLoaderParameter parameter) {
+        // First make sure the current is cleared
+        maramafication = null;
+
+        // Since the maramafications are currently saved as json files, read the json file
+        // This should be altered when there is another way of passing maramafication information
```
Comment (LuneCoding) [inline core/src/com/marama/view/util/MaramaficationLoader.java:37]: Backlog card if you think this is something that ought to be done.

File: core/src/com/marama/view/renderables/World.java [hunk 5]
```diff
@@ -162,22 +167,43 @@ private void init() {
...
     private void doneLoading() {
-        // Create a nice 3D grid of MBlocks.
-        for (float x = -3f; x <= 3f; x += 2f) {
-            for (float z = -3f; z <= 3f; z += 2f) {
-                for (float y = -3f; y <= 3f; y += 2f) {
-                    SelectableInstance instance = mBlock.createInstance();
-                    instance.transform.setToTranslation(x, y, z);
-                    modelInstances.add(instance);
-                }
+//        this.maramaBlock.transform.setToScaling(3.0f, 3.0f, 3.0f);
```
Comment (LuneCoding) [inline core/src/com/marama/view/renderables/World.java]: Why all this commented code?

File: core/src/com/marama/view/renderables/World.java [hunk 6]
```diff
@@ -13,21 +14,25 @@
...
     private DirectionalLight directionalLight;
-    private Array<ModelInstance> modelInstances; // All models that are in the World.
     private PerspectiveCamera perspectiveCamera;
+    private ModelBatch modelBatch; // The unit that can render the modelInstances
     private CameraInputController cameraInputController;
-    private AssetManager assetManager;
-    private ModelBatch modelBatch;
-    private MBlock mBlock;
+
+    private EntityManagerSingleton entityManager; // The unit that can hold all the models
```
Comment (LuneCoding) [inline core/src/com/marama/view/renderables/World.java]: Comment really doesn't explain anything.

Comment (jjkramok) [reply core/src/com/marama/view/renderables/World.java]: This comment is gold.

File: core/src/com/marama/view/util/MaramaficationLoader.java (hunk 2 above)
Comment (ThijsvanDam) [reply core/src/com/marama/view/util/MaramaficationLoader.java:66]: Nope. This is just required for an AsynchronousAssetLoader.

File: core/src/com/marama/view/util/MaramaficationLoader.java (hunk 3 above)
Comment (ThijsvanDam) [reply core/src/com/marama/view/util/MaramaficationLoader.java:63]: Nope. This is just required for an AsynchronousAssetLoader.

File: core/src/com/marama/view/util/MaramaficationLoader.java (hunk 4 above)
Comment (ThijsvanDam) [reply core/src/com/marama/view/util/MaramaficationLoader.java:37]: This'll automatically be done if we can communicate with scala. I can create a backlog item, though.

File: core/src/com/marama/view/renderables/World.java (hunk 5 above)
Comment (ThijsvanDam) [reply core/src/com/marama/view/renderables/World.java]: removed

**Review by wilbo — APPROVED** [review]

**Review by LuneCoding — APPROVED** [review]

**Pull request merged**